pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -419,6 +413,12 @@ func (d *DeploymentController) apply(controller string, yml string) error {
 	if err != nil {
 		return err
 	}
+	canManage, resourceVersion := d.canManage(gvr, us.GetName(), us.GetNamespace())
+	if !canManage {
+		log.Debugf("skipping %v/%v/%v, already managed", gvr, us.GetName(), us.GetNamespace())
+		return nil
```

Comment:
In this case, we should update the status to reflect that the gateway has not been implemented.  That might be happing elsewhere, but I can't tell from reading the code...